pkgs/json_rpc_2/lib/src/client.dart

```diff
@@ -18,16 +18,18 @@ import 'utils.dart';
 class Client {
   final StreamChannel<dynamic> _channel;
 
-  /// The next request id.
-  var _id = 0;
+  /// A function to generate the next request id.
+  Object Function() _generateId;
 
   /// The current batch of requests to be sent together.
   ///
   /// Each element is a JSON RPC spec compliant message.
   List<Map<String, dynamic>>? _batch;
 
   /// The map of request ids to pending requests.
-  final _pendingRequests = <int, _Request>{};
+  ///
+  /// Keys must be of type `int` or `String`.
+  final _pendingRequests = <Object, _Request>{};
```

Comment:
perhaps:

```suggestion
  final Map<Object, _Request> _pendingRequests = {};
```

Reply:
That doesn't match the style elsewhere in this package, so if we want to do that I would treat it as a separate refactor on the whole package.